### PR TITLE
Modified await statement to remove compiler warning while awaiting on the file stream to be created.

### PR DESCRIPTION
**Pull request by jbatt33**
Working on compiler warning for this line of code. 

**Changes**
The changes to how the file stream is declared and added some brackets for the using.

The using will still be in the main context, while the flush will happen outside.

### CODE REVIEW

File: MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs [hunk 1]
```diff
@@ -23,8 +23,12 @@ public async Task<string> WriteDocumentAsync(string clientName, string clientVer
         {
             var fileName = $"upload_{clientName}_{clientVersion}_{DateTime.UtcNow:yyyyMMddHHmmss}_{Guid.NewGuid():N}.log";
             var logFilePath = Path.Combine(_applicationPaths.LogDirectoryPath, fileName);
-            await using var fileStream = new FileStream(logFilePath, FileMode.CreateNew, FileAccess.Write, FileShare.None);
-            await fileContents.CopyToAsync(fileStream).ConfigureAwait(false);
+            FileStream fileStream;
+            await using (fileStream = new FileStream(logFilePath, FileMode.CreateNew, FileAccess.Write, FileShare.None))
```
Comment (IDisposable) [inline MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs:27]: `await using (var fileStream = new FileStream(logFilePath, FileMode.CreateNew, FileAccess.Write, FileShare.None))` would be better... no reason for distinct declaration.

File: MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs [hunk 2]
```diff
@@ -23,8 +23,12 @@ public async Task<string> WriteDocumentAsync(string clientName, string clientVer
         {
             var fileName = $"upload_{clientName}_{clientVersion}_{DateTime.UtcNow:yyyyMMddHHmmss}_{Guid.NewGuid():N}.log";
             var logFilePath = Path.Combine(_applicationPaths.LogDirectoryPath, fileName);
-            await using var fileStream = new FileStream(logFilePath, FileMode.CreateNew, FileAccess.Write, FileShare.None);
-            await fileContents.CopyToAsync(fileStream).ConfigureAwait(false);
```
Comment (IDisposable) [inline MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs:27]: This seems to be a common anti-pattern... the `using` line should always have braces enclosing the body using the `IDisposable` object, that's the entire purpose OF the using statement. Seems that when used here, we're seeing a false understanding that the next line is the scope of the using, rather it is "anytime you want, but before the next-outer scope close" is the real spec.

I think we need to make a sweep looking for this (I just fixed an issue caused by this error in PR #9559), and I might be the right person to do it :)

Comment (crobibero) [reply MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs:27]: This isn't an anti-pattern but was a feature added in C# 8.
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using

Comment (IDisposable) [reply MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs:27]: It does cause the class of bugs I fixed in PR #9559.

File: MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs (hunk 1 above)
Comment (jbatt33) [reply MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs:27]: I was not able to resolve the warning without writing it this way.

File: MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs (hunk 2 above)
Comment (jbatt33) [reply MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs:27]: Looks like some good reading here for me.

**Comment by Bond-009** [comment]
I think this isn't needed anymore due to #10390

**Pull request closed without merging**